specs/TransportProtocols.md

```diff
@@ -0,0 +1,175 @@
+# Transport Protocols
+
+This document describes the protocols used by the three ASP.NET Endpoint Transports: WebSockets, Server-Sent Events, Long Polling and HTTP Post
+
+## Transport Requirements
+
+A transport is required to have the following attributes:
+
+1. Duplex - Able to send messages from Server to Client and from Client to Server
+1. Frame-based - Messages have fixed length (as opposed to streaming, where data is just pushed throughout the connection)
+1. Frame Metadata - Able to encode two pieces of Frame Metadata
+    * `Type` - Either `Binary` or `Text`.
+    * `EndOfMessage` - Boolean indicating if this frame represents the final frame of a complete message.
```

Comment:
Ping @davidfowl - I'm leaning really close to just saying that only WebSockets supports "partial" messages. The big trick is what do we do with EndPoints that provide Message objects with EndOfMessage set to false, when the transport doesn't support them. Buffer?